src/scene_battle_rpg2k3.cpp

```diff
@@ -2083,6 +2089,65 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 		FaceTarget(*actor, *action->GetTarget());
 	}
 
+	if (action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
+		SetBattleActionState(BattleActionState_MoveForward);
+	} else {
+		SetBattleActionState(BattleActionState_StartAlgo2);
+	}
+	return BattleActionReturn::eContinue;
+}
+
+Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleActionMoveForward(Game_BattleAlgorithm::AlgorithmBase* action) {
+	auto* source = action->GetSource();
+
+	if (source->GetType() == Game_Battler::Type_Ally) {
+		auto* target = action->GetTarget();
+
+		if (cba_move_frame == 0) {
+			auto* actor = static_cast<Game_Actor*>(source);
+			auto* sprite = actor->GetActorBattleSprite();
+			if (sprite) {
+				sprite->SetAnimationState(Sprite_Actor::AnimationState_WalkingLeft);
+			}
+			cba_start_pos = source->GetBattlePosition();
+			if (action->GetCBAMovement() == lcf::rpg::BattlerAnimationItemSkill::Movement_move) {
+				if (target != nullptr) {
+					auto* enemy = static_cast<Game_Enemy*>(target);
+					auto* enemysprite = enemy->GetEnemyBattleSprite();
+					if (enemysprite) {
+						cba_end_pos = Point(target->GetBattlePosition().x + (source->IsDirectionFlipped() ? -(enemysprite->GetWidth() / 2) : enemysprite->GetWidth() / 2), target->GetBattlePosition().y);
+					}
+				}
+			}
+		}
+
+		if (cba_move_frame < 15) {
```

Comment:
15 frames needs confirmation in RE tools. Leaving a note

Reply:
dito (comment)

Reply:
Added FIXME comment.